lib/IRGen/GenMeta.cpp

```diff
@@ -3837,13 +3839,21 @@ namespace {
   public:
     ForeignClassMetadataBuilder(IRGenModule &IGM, ClassDecl *target,
                                 ConstantStructBuilder &B)
-      : ForeignMetadataBuilderBase(IGM, target, B) {}
+        : ForeignMetadataBuilderBase(IGM, target, B) {
+      if (IGM.getOptions().LazyInitializeClassMetadata)
```

Comment:
This looks like it should be part of a needsForeignMetadataCompletionFunction() check elsewhere

Reply:
Okay, I'll take a look and see if I can fold that in.